util/src/main/scala/Money.scala

```diff
@@ -253,17 +261,46 @@ object Money {
   def GBP(amount: BigDecimal): Money = decimalAmountWithCurrencyAndHalfEvenRounding(amount, "GBP")
   def JPY(amount: BigDecimal): Money = decimalAmountWithCurrencyAndHalfEvenRounding(amount, "JPY")
 
-  val CurrencyCodeField: String = "currencyCode"
-  val CentAmountField: String = "centAmount"
-  val FractionDigitsField: String = "fractionDigits"
-  val TypeName: String = "centPrecision"
+  final val CurrencyCodeField: String = "currencyCode"
+  final val CentAmountField: String = "centAmount"
+  final val FractionDigitsField: String = "fractionDigits"
+  final val TypeName: String = "centPrecision"
 
   def fromDecimalAmount(amount: BigDecimal, currency: Currency)(implicit
-      mode: RoundingMode): Money =
-    Money(amount.setScale(currency.getDefaultFractionDigits, mode), currency)
+      mode: RoundingMode): Money = {
+    val fractionDigits = currency.getDefaultFractionDigits
+    val centAmountBigDecimal = amount * cachedCentPower(fractionDigits)
+    val centAmount = centAmountBigDecimal.setScale(0, mode).longValue
+    Money(centAmount, currency)
+  }
+
+  def apply(amount: BigDecimal, currency: Currency): Money = {
+    require(
+      amount.scale == currency.getDefaultFractionDigits,
+      "The scale of the given amount does not match the scale of the provided currency." +
+        " - " + amount.scale + " <-> " + currency.getDefaultFractionDigits
+    )
+    fromDecimalAmount(amount, currency)(BigDecimal.RoundingMode.HALF_EVEN)
```

Comment:
I'm trying to understand why choosing the `RoundingMode.HALF_EVEN`.
Given that we know the scale of `amount` matches the fraction digits of the currency, would it make a difference if the rounding mode would be `RoundingMode.UNNECESSARY`, as in old `fromCentAmount` method?

Reply:
Thanks for catching my attention on this.
Indeed, I also think now that `UNNECESSARY` is suitable here, like it was used in https://github.com/commercetools/sphere-scala-libs/blob/main/util/src/main/scala/Money.scala#L289
I'll have a deeper look.

Reply:
I can confirm
https://github.com/commercetools/sphere-scala-libs/pull/375/commits/9b1f5023074093a2adb6a799543b0c9dc5568ebb